tests/Set/IntersectionTest.php

```diff
@@ -0,0 +1,630 @@
+<?php
+
+declare(strict_types=1);
+
+namespace IterTools\Tests\Set;
+
+use IterTools\Set;
+use IterTools\Tests\Fixture\ArrayIteratorFixture;
+use IterTools\Tests\Fixture\GeneratorFixture;
+use IterTools\Tests\Fixture\IteratorAggregateFixture;
+
+class IntersectionTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * @dataProvider dataProviderForArray
+     * @param        array<array> $iterables
+     * @param        array $expected
+     */
+    public function testArray(array $iterables, array $expected): void
+    {
+        // Given
+        $result = [];
+
+        // When
+        foreach (Set::intersection(...$iterables) as $datum) {
+            $result[] = $datum;
+        }
+
+        // Then
+        $this->assertEquals($expected, $result);
+    }
+
+    public function dataProviderForArray(): array
+    {
+        return [
+            [
+                [],
+                [],
+            ],
+            [
+                [
+                    [],
+                ],
+                [],
+            ],
+            [
+                [
+                    [],
+                    [],
+                ],
+                [],
+            ],
+            [
+                [
+                    [],
+                    [],
+                    [],
+                ],
+                [],
+            ],
+            [
+                [
+                    [1, 2, 3, 4, 5],
+                    [1, 2, 3, 4, 5],
+                ],
+                [1, 2, 3, 4, 5],
+            ],
+            [
+                [
+                    [1, 2, 3, 4, 5],
+                    [1, 2, 3, '4', '5'],
+                ],
+                [1, 2, 3],
+            ],
+            [
+                [
+                    ['1', '2', '3', 4, 5],
+                    [1, 2, 3, '4', '5'],
+                ],
+                [],
+            ],
+            [
+                [
+                    [1, 1, 2, 2, 1, 1],
+                    [2, 2, 1, 1, 2, 2],
+                ],
+                [2, 1, 2, 1],
```

Comment:
I should have asked this before merging it. Can you explain this unit test to me? I'd expect a set intersection to result in [1, 2].

Reply:
As we know, a set must contain unique values. If it contains duplicate values, then it is a multiset.

We cannot guarantee that only unique values will be in the iterable, so I provided the logic for operations with multisets.

Based on this article: https://trizenx.blogspot.com/2016/05/multisets.html?m=1

Reply:
OK thanks for the clarification. Multisets is the answer. Thanks.

Reply:
Thank you for your vigilance and care.

Reply:
Follow-up question: Is the order important? The unit test is testing the order, but would [1, 1, 2, 2] or [1, 2, 1, 2] also be acceptance answers to the intersection of these two multisets?

Reply:
I think order is not important. So we can presort the result before asserting, if you like this approach.

Reply:
There is no need to sort. PHPUnit has a method [`assertEqualsCanonicalizing`](https://phpunit.readthedocs.io/en/9.5/assertions.html#assertequalscanonicalizing) that will assert the values in any order. I will update the merged tests.

Reply:
Thank you Mark!